airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/csv_parser.py

```diff
@@ -137,10 +142,21 @@ def infer_schema_process(
         # and so we can't multiprocess with the actual fileobject on Windows systems
         # we're reading block_size*2 bytes here, which we can then pass in and infer schema from block_size bytes
         # the *2 is to give us a buffer as pyarrow figures out where lines actually end so it gets schema correct
-        file_sample = file.read(self._read_options()["block_size"] * 2)
-        schema_dict = self._run_in_external_process(
-            infer_schema_process, 4, 60, file_sample, self._read_options(), self._parse_options(), self._convert_options()
-        )
+        if self._format.get("infer_datatypes", True):
+            self.logger.debug("inferring schema")
+            file_sample = file.read(self._read_options()["block_size"] * 2)
+            schema_dict = self._run_in_external_process(
+                infer_schema_process, 4, 60, file_sample, self._read_options(), self._parse_options(), self._convert_options()
+            )
+        else:
+            self.logger.debug("infer_datatypes is False, skipping infer_schema")
+            encoding = self._format.get("encoding", "UTF-8")
+            delimiter = self._format.get("delimiter", ",")
+            quote_char = self._format.get("quote_char")
+            reader = csv.reader([file.readline().decode(encoding)], delimiter=delimiter, quotechar=quote_char)
```

Comment:
pycharm complains about str not having `decode` method, are you sure this will work with both `IOText` and `BinaryIO`?

Reply:
added a call to six.ensure_text